chain-impl-mockchain/src/vote/tally.rs

```diff
@@ -21,15 +21,38 @@ pub struct TallyResult {
     options: Options,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, Eq, PartialEq)]
 pub enum Tally {
     Public { result: TallyResult },
+    Private { state: PrivateTallyState },
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum PrivateTallyState {
+    Encrypted {
+        encrypted_tally: chain_vote::Tally,
+        total_stake: Stake,
+    },
+    Decrypted {
+        result: TallyResult,
+    },
 }
 
 #[derive(Debug, Error, PartialEq, Eq, Clone)]
 pub enum TallyError {
     #[error("Invalid option choice")]
     InvalidChoice { options: Options, choice: Choice },
+    #[error("Invalid privacy")]
+    InvalidPrivacy,
+    #[error("Choice number in private vote was too high. Maximum of 255 is allowed.")]
+    InvalidPrivateChoiceSize {
+        #[from]
+        source: std::num::TryFromIntError,
```

Comment:
nitpick: this is too baroque, an integer member would be lighter and you could use it in the display message.